### ISSUE TITLE
`rasa interactive` stuck in infinite loop in Rasa 3.0rc1

### ISSUE THREAD

**Issue by koaning** [RasaHQ/rasa#10213]
I'm trying to teach my assistant to handle a form correctly. In particular, I have a form that has these slots: 

```yaml
forms:
  fancy_pizza_form:
    required_slots:
    - vegetarian
    - pizza_size
    - pizza_type
slots:
  pizza_size:
    type: text
    influence_conversation: true
    mappings:
    - type: from_entity
      entity: pizza_size
  pizza_type:
    type: text
    influence_conversation: true
    mappings:
    - type: from_entity
      entity: pizza_type
  vegetarian:
    type: bool
    influence_conversation: true
    mappings:
    - type: from_intent
      value: true
      intent: affirm
    - type: from_intent
      value: false
      intent: deny
```

I don't have stories for this form yet, so I figured I'd add one via `rasa interactive`. I'm stuck at this step. 

```
------
Chat History

 #    Bot                                                                You        
────────────────────────────────────────────────────────────────────────────────────
 1    action_listen                                                                 
────────────────────────────────────────────────────────────────────────────────────
 2                                                                         hi there 
                                                                 intent: greet 1.00 
────────────────────────────────────────────────────────────────────────────────────
 3    utter_greet 1.00                                                              
      Hey! How are you?                                                             
      action_listen 1.00                                                            
────────────────────────────────────────────────────────────────────────────────────
 4                                                      i want to buy a fancy pizza 
                                                       intent: buy_fancy_pizza 1.00 
────────────────────────────────────────────────────────────────────────────────────
 5    fancy_pizza_form 1.00                                                         
      active_loop{"name": "fancy_pizza_form"}                                       
      slot{"requested_slot": "vegetarian"}                                          
      Would you like to order a vegetarian pizza?                                   
      Buttons:                                                                      
      1: yes (/affirm)                                                              
      2: no (/deny)                                                                 
      Type out your own                                                             
      message...                                                                    
      action_listen 1.00                                                            
────────────────────────────────────────────────────────────────────────────────────
 6                                                                            /deny 
                                                                  intent: deny 1.00 


Current slots: 
        pizza_size: None, pizza_type: None, vegetarian: None, requested_slot: vegetarian, session_started_metadata: None

------
? The bot wants to run 'action_default_fallback', correct? No                                                                                      
------
```

There's a fallback triggered! That's no bueno. So I'll try to correct it. 

```
? What is the next action of the bot? (Use arrow keys)                                                                                             
   <create new action>
   0.30 action_default_fallback                 
   0.00 ...                                     
   0.00 action_ask_pizza_type                   
   0.00 action_ask_vegetarian                   
   0.00 action_back                             
   0.00 action_deactivate_loop                  
   0.00 action_default_ask_affirmation 
...
 » 0.00 fancy_pizza_form                        
```

This brings up a prompt. So far so good. 

```
? What is the next action of the bot? 0.00 fancy_pizza_form                                                                                        
Thanks! The bot will now run fancy_pizza_form.

? Should 'fancy_pizza_form' validate user input to fill the slot 'vegetarian'? (Y/n)
```

I hit `Y` and I see ... 

```
Chat History

 #    Bot                                                                You        
────────────────────────────────────────────────────────────────────────────────────
 1    action_listen                                                                 
────────────────────────────────────────────────────────────────────────────────────
 2                                                                         hi there 
                                                                 intent: greet 1.00 
────────────────────────────────────────────────────────────────────────────────────
 3    utter_greet 1.00                                                              
      Hey! How are you?                                                             
      action_listen 1.00                                                            
────────────────────────────────────────────────────────────────────────────────────
 4                                                      i want to buy a fancy pizza 
                                                       intent: buy_fancy_pizza 1.00 
────────────────────────────────────────────────────────────────────────────────────
 5    fancy_pizza_form 1.00                                                         
      active_loop{"name": "fancy_pizza_form"}                                       
      slot{"requested_slot": "vegetarian"}                                          
      Would you like to order a vegetarian pizza?                                   
      Buttons:                                                                      
      1: yes (/affirm)                                                              
      2: no (/deny)                                                                 
      Type out your own                                                             
      message...                                                                    
      action_listen 1.00                                                            
────────────────────────────────────────────────────────────────────────────────────
 6                                                                            /deny 
                                                                  intent: deny 1.00 


Current slots: 
        pizza_size: None, pizza_type: None, vegetarian: None, requested_slot: vegetarian, session_started_metadata: None

------
? The bot wants to run 'action_default_fallback', correct? (Y/n)
```

Crud ... stuck in a loop. It feels like there's a bug around here. 

```
> python -m rasa --version
Rasa Version      :         3.0.0rc1
Minimum Compatible Version: 2.8.9
Rasa SDK Version  :         3.0.0rc1
Rasa X Version    :         None
Python Version    :         3.7.9
Operating System  :         Linux-5.11.0-7614-generic-x86_64-with-Pop-20.10-groovy
Python Path       :         /home/vincent/Development/how-to-rasa/venv/bin/python
```

**Comment by koaning** [comment]
I just tried again after adding a story using the form in the `stories.yml` file. The issue seems to persist.

**Comment by koaning** [comment]
@ancalita to make it easier to reproduce, I've created a repo with the Rasa project [here](https://github.com/RasaHQ/issue-10213).

**Comment by koaning** [comment]
I can confirm that this issue does not appear in `rasa shell`. It indeed seems like the culprit is `rasa interactive` and that it is unable to set slots.

**Comment by koaning** [comment]
It's very likely that my experience is related to this issue: https://github.com/RasaHQ/rasa/issues/10220.

**Comment by ancalita** [comment]
@koaning This issue seems to be fixed by this PR #10252 by @ka-bu  🎉 
I've just tested something similar on the PR branch and there's no more fallback prediction 😄

**Comment by ka-bu** [comment]
Just tested again (with form as above + entities + nlu examples for new inform intent and the entities + creating buy_fancy_pizza intent on the fly) and can confirm that there's no fallback anymore => Will be closed via #10252 

However, also observed the issue from #10212, so we're not quite there yet 😬
